lib/dep-graph-builders/util.ts

```diff
@@ -0,0 +1,64 @@
+import { PackageJsonBase } from './types';
+import { DepGraphBuilder } from '@snyk/dep-graph';
+
+export interface PkgNode {
+  id: string;
+  name: string;
+  version: string;
+  dependencies: Record<string, { version: string; isDev: boolean }>;
+  isDev: boolean;
+}
+
+export const addPkgNodeToGraph = (
+  node: PkgNode,
+  isCyclic: boolean,
+  depGraphBuilder: DepGraphBuilder,
+): void => {
+  depGraphBuilder.addPkgNode(
+    { name: node.name, version: node.version },
+    node.id,
+    {
+      labels: {
+        scope: node.isDev ? 'dev' : 'prod',
+        ...(isCyclic && { pruned: 'cyclic' }),
+      },
+    },
+  );
+};
+
+/**
+ * Get top level dependencies from the given package json object which is parsed from a package.json file.
+ * This includes both prod dependencies and dev dependencies supposing includeDevDeps is supported.
+ */
+export const getTopLevelDeps = (
+  pkgJson: PackageJsonBase,
+  options: { includeDevDeps: boolean },
+): Record<string, { version: string; isDev: boolean }> => {
+  const prodDeps = getGraphDependencies(pkgJson.dependencies || {}, false);
+
+  const devDeps = options.includeDevDeps
+    ? getGraphDependencies(pkgJson.devDependencies || {}, true)
+    : {};
+
+  return { ...prodDeps, ...devDeps };
+};
+
+/**
+ * Converts dependencies parsed from the yarn.lock file to a dependencies object required by the graph.
```

Comment:
yarn.lock is incorrect here right? Otherwise this isn't a general function

Reply:
this is so well spotted :))) thank you :)))